### ISSUE TITLE
Forward declarations

### ISSUE THREAD

**Issue by zerodefect** [jupp0r/prometheus-cpp#190]
I often find myself needing to declare forward declarations for the 'Registry' class (or even for one of the different metric types). Would it be possible to include a `registry_fwd.h` this as part of the project itself?

**Comment by jerryct** [comment]
Why not simply include the header as recommended by https://google.github.io/styleguide/cppguide.html#Forward_Declarations ?

**Comment by zerodefect** [comment]
I'm familiar with that guide. As in the name, it's just a 'guide'  

For me compile times are very important - especially when building large projects. As an example, I build my project in the cloud -  I pay per minute of build time. 

Now the registry.h file includes a lot of headers which in some situations are bloat.  I do think that including forward declarations would not be onerous to the project maintainers and in fact, could be merely ignored by those who choose not to use them.

**Comment by jovan-wong** [comment]
I agree that we follow the google code style as such an open project.

**Comment by jupp0r** [comment]
Why not create your own `prometheus_fwd.h` in your project if you need to forward-declare types and want to avoid duplication? I can see your point, but I think it should be the libraries client's concern because different clients have different requirements (you might want to forward-declare other types than registry, for example).
